syncer/db.go

```diff
@@ -196,45 +251,40 @@ func (conn *Conn) executeSQLWithIgnore(tctx *tcontext.Context, ignoreError func(
 	return ret.(int), nil
 }
 
-func (conn *Conn) executeSQL(tctx *tcontext.Context, queries []string, args ...[]interface{}) (int, error) {
+func (conn *WorkerConn) executeSQL(tctx *tcontext.Context, queries []string, args ...[]interface{}) (int, error) {
 	return conn.executeSQLWithIgnore(tctx, nil, queries, args...)
 }
 
-func createBaseConn(dbCfg config.DBConfig, timeout string) (*baseconn.BaseConn, error) {
-	dbDSN := fmt.Sprintf("%s:%s@tcp(%s:%d)/?charset=utf8mb4&interpolateParams=true&readTimeout=%s&maxAllowedPacket=%d",
-		dbCfg.User, dbCfg.Password, dbCfg.Host, dbCfg.Port, timeout, *dbCfg.MaxAllowedPacket)
-	baseConn, err := baseconn.NewBaseConn(dbDSN, &retry.FiniteRetryStrategy{})
+func createConn(ctx context.Context, cfg *config.SubTaskConfig, dbCfg config.DBConfig) (*conn.BaseDB, *WorkerConn, error) {
+	baseDB, err := createBaseDB(dbCfg)
 	if err != nil {
-		return nil, terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+		return nil, nil, err
 	}
-	return baseConn, nil
-}
-
-func createConn(cfg *config.SubTaskConfig, dbCfg config.DBConfig, timeout string) (*Conn, error) {
-	baseConn, err := createBaseConn(dbCfg, timeout)
+	baseConn, err := baseDB.GetBaseConn(ctx)
 	if err != nil {
-		return nil, err
+		return nil, nil, terror.WithScope(terror.DBErrorAdapt(err, terror.ErrDBDriverError), terror.ScopeDownstream)
 	}
-	return &Conn{baseConn: baseConn, cfg: cfg}, nil
+	return baseDB, &WorkerConn{baseDB: baseDB, baseConn: baseConn, cfg: cfg}, nil
```

Comment:
the `WorkerConn` contains baseDB, why need return baseDB?

Reply:
It seems strange that `WorkerConn` has `BaseDB`, because one `BaseDB` could generate multiple `WorkerConn`, so I put `BaseDB` outside.
but if `WorkerConn` need to re-establish its connection, it must use `BaseDB`, so put `BaseDB` back in `WorkerConn` can re-establish itself. which one is better? Do you have any suggestions? to put it inside or outside

Reply:
ok, I don't have better suggections